pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2233,6 +2263,13 @@ private void updateBrokerPublisherThrottlingMaxRate() {
         }
     }
 
+    private void updateCacheSizeAndThreshold(long maxSize) {
+        ManagedLedgerFactoryImpl managedLedgerFactory = (ManagedLedgerFactoryImpl) pulsar().getManagedLedgerFactory();
+        double thresholdPercent = managedLedgerFactory.getEntryCacheManager().getEvictionTriggerThresholdPercent();
+        managedLedgerFactory.getEntryCacheManager().setMaxSize(maxSize);
```

Comment:
We should move these lines into  method in ManagedLedgerFactoryImpl
Otherwise we are breaking incapsulation

Probably it is better to add the new method to the interface and not only in the Impl classe

Reply:
The same applies to setCacheEvictionTimeThresholdNanos

Reply:
OK I will fix

Reply:
Fixed